tests/test_creator.cairo

```diff
@@ -0,0 +1,590 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import (HashBuiltin,
+    BitwiseBuiltin)
+
+from contracts.creator import (create, contribute, user_counts, generation_of_game, view_game, newest_game)
+
+@external
+func test_create_happy_case{
+        syscall_ptr : felt*,
+        bitwise_ptr : BitwiseBuiltin*,
+        pedersen_ptr : HashBuiltin*,
+        range_check_ptr
+    }():
+    alloc_locals
+    local game_state = 1
+    local game_hash = 2001140082530619239661729809084578298299223810202097622761632384561112390979
+    %{
+        expect_events({"name": "game_created", "data": [123, 2, ids.game_hash, 1]})
+        expect_events({"name": "credit_reduced", "data": [123, 0]})
+    %}
+
+    %{ stop_prank_callable = start_prank(123) %}
+    # add credits by progressing 10 games
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+    contribute(game_index=1)
+
+    let (_, credit_count) = user_counts(user_id=123)
+    assert credit_count = 10
+
+    # create new game
+    create(game_state=game_state)
+
+    %{ stop_prank_callable() %}
+
+    # check if credits were reduced and game was accounted
+    let (user_games, new_credit_count) = user_counts(user_id=123)
+    assert new_credit_count = 0
+    assert user_games = 1
+
+    # check index
+    let (game_index, game_id, generation) = newest_game()
+    assert game_index = 2
+    assert game_id = game_hash
+    generation = 0
+
+    # check the game is properly stored
+    let (game_len, game) = view_game(game_index=game_index, generation=generation)
```

Comment:
I think that if view_game returned packed game the state assertions would be easier (one value is needed and python can be used to generate it).